File: Design-Patterns/Decorator/Cpp/attributes.cpp
#include "attributes.h"

attributes::~attributes() = default;

int attributes::get_max_mana()
{
	return max_mana_;
}

int attributes::get_max_health()
{
	return max_health_;
}

int attributes::get_attack()
{
	return attack_;
}

int attributes::get_defense()
{
	return defense_;
}

int attributes::get_critical()
{
	return critical_;
}

File: Design-Patterns/Decorator/CSharp/Attributes/Attributes.cs
namespace Decorator.Attributes
{
    public class Attributes
    {
        public virtual int MaxHealth { get; protected set; }
        public virtual int MaxMana { get; protected set; }
        public virtual int Attack { get; protected set; }
        public virtual int Defense { get; protected set; }
        public virtual int Critical { get; protected set; }
    }
}
